### DISCUSSION TITLE
no way to read pulses

### DISCUSSION THREAD

**Discussion by pielle63** [HomeSpan/HomeSpan#533] [Q&A]
Hi,
I’m using two ESP32 Pulse Counter peripherals (PCNT_UNIT_0 and PCNT_UNIT_1)  to count some pulses at 50Hz, using the library <pcnt.h>.  HW and SW are properly initialised and the SW reads the PCNT values using ‘pcnt_get_counter_value’.
PCNT_UNIT_1 works properly while PCNT_UNIT_0 returns random numbers between 0  and 20.  
Removing homeSpan.poll() from loop() the counting works perfectly on both channels
This problem popped up after a massive update of libraries but unfortunately I’m unable to move back to the previous configuration

What I tried up to now:
-	read twice each PCNT register, saving the results in two different  memory locations (to exclude memory corruption) : same behaviour = the data in the PCNT registers are corrupted.
- run homeSpan.poll() at 10Hz instead in the loop(), immediately after the function reading PCNT register: same behaviour
-	change to different PCNT units: same behaviour
-	replace counting implementation from PCNT units to interrupts: each pulse generates an interrupt and the service routine increase a counter: same behaviour

Now I’m in short of ideas.  Just one idea is left: move all to FreeRTOS in order to isolate HomeSpan in one core and the other functions in the other core. 
What do you think?  could works?
Any further suggestion?

Thank you in advance for any suggestion

**Comment by HomeSpan** [comment]
Try running HomeSpan in background with homeSpan.autopoll().

**Reply by pielle63** [reply]
thank you for your suggestion Homespan, but unfortunately the system behaves the same way even increasing the stackSize to 16K.
based on this results, does it make sense to move to FreeRTOS?
any other suggestion?
thanks!

**Comment by HomeSpan** [comment]
If the library previously worked but stopped working after an update, there may be some aspects that are directly interfering with HomeSpan.  Hard to say if moving everything to FreeRTOS will solve the issue without knowing exactly what is causing the problem.

**Reply by pielle63** [reply]
Solved: was a wrong HW configuration.
